### ISSUE TITLE
Creating custom layer in java Deeplearning4J

### ISSUE THREAD

**Issue by ghost** [deeplearning4j/deeplearning4j#9152]
I have some problems importing my custom layer into my Java project. I read the documentation (https://deeplearning4j.konduit.ai/keras-import/custom-layers) but without succes. I've seen the two examples but it isn't clear for me how to **exactly** implement my method in Java, like what should I implement etc?

This is the python code:
```
class lstm_bottleneck(tf.keras.layers.Layer):
    def __init__(self, lstm_units, time_steps, **kwargs):
        self.lstm_units = lstm_units
        self.time_steps = time_steps
        self.lstm_layer = Bidirectional(LSTM(lstm_units, return_sequences=False))
        self.repeat_layer = RepeatVector(time_steps)
        super(lstm_bottleneck, self).__init__(**kwargs)

    def call(self, inputs):
        # just call the two initialized layers
        return self.repeat_layer(self.lstm_layer(inputs))

    def compute_mask(self, inputs, mask=None):
        # return the input_mask directly
        return mask

    def get_config(self):
        thisDict = {
            "lstm_layer": self.lstm_layer.get_config(),
            "repeat_layer": self.repeat_layer.get_config()
        }
        return thisDict
```

And this is my Java code. Ofcourse it isn't working because i'm just trying to get the examples given in the documentation... (lstm_units and time_steps are fixed numbers so i guess i shouldn't implement that in the get_config() method).
```
public class lstm_bottleneck extends KerasLayer {

    private int lstm_units = 45;
    private int time_steps = 322;

    public lstm_bottleneck(Map<String, Object> layerconfig)
            throws InvalidKerasConfigurationException, UnsupportedKerasConfigurationException {
        this(layerconfig, true);
    }

    public lstm_bottleneck(Map<String, Object> layerconfig, boolean enforceTrainingConfig)
            throws InvalidKerasConfigurationException, UnsupportedKerasConfigurationException{
        super(layerconfig, enforceTrainingConfig);

        Map<String, Object> lrnParams = KerasLayerUtils.getInnerLayerConfigFromConfig(layerconfig, conf);

    }

}

```

Anyone who can help? Thanks in advance!!

**Comment by ghost** [comment]
@treo Could you by any chance check this out? Sorry for the pressure, but it's for my master's thesis and the deadline is next week. So it would really be a huge help!

**Comment by agibsonccc** [comment]
@DataPlanningEngineer  you will need to write forward and backwards yourself unless you use SameDiff inside. https://github.com/eclipse/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/samediff/SameDiffLambdaLayer.java

We have a custom layer examples here:  https://github.com/eclipse/deeplearning4j-examples/tree/master/dl4j-examples/src/main/java/org/deeplearning4j/examples/advanced/features/customizingdl4j

Here's loss functions, layers above.

For keras, here's the tests for that:
https://github.com/eclipse/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/e2e/KerasCustomLayerTest.java

The associated mentioned layers can be found here:
https://github.com/eclipse/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-modelimport/src/main/java/org/deeplearning4j/nn/modelimport/keras/layers/custom/KerasPoolHelper.java

**Comment by agibsonccc** [comment]
Closing due to lack of reply
